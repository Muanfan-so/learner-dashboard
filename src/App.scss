// frontend-app-*/src/index.scss
@import "~@edx/brand/paragon/fonts";
@import "~@edx/brand/paragon/variables";
@import "~@edx/paragon/scss/core/core";
@import "~@edx/brand/paragon/overrides";

$fa-font-path: "~font-awesome/fonts";
@import "~font-awesome/scss/font-awesome";

$input-focus-box-shadow: $input-box-shadow; // hack to get upgrade to paragon 4.0.0 to work

@import "~@edx/frontend-component-footer/dist/_footer";

.alert .alert-icon {
  color: black;
}

#root {
  // Removing a odd 1.5 scaling on checkboxes.:
  input[type=checkbox] {
    transform: none;
  }
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  main {
    flex-grow: 1;
  }

  header {
    flex: 0 0 auto;

    .logo {
      display: block;
      box-sizing: content-box;
      position: relative;
      top: 0.1em;
      height: 1.75rem;
      margin-right: 1rem;
      img {
        display: block;
        height: 100%;
      }
    }
  }

  footer {
    flex: 0;
  }
}

#paragon-portal-root {
  .pgn__modal-layer {
    .pgn__modal-close-container {
      right: 1rem !important;
    }
  }
  .confirm-modal .pgn__modal-body {
    overflow: hidden;
  }
}

.courses-panel {
  margin-top: 65px;
}

.recommended-course-card {
  .pgn__card-wrapper-image-cap {
    max-width: 115.01px !important;
    min-width: 115.01px !important;
    background: #FFFFFF;
    box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.15), 0px 1px 4px rgba(0, 0, 0, 0.15);
    border-radius: 4px;
    padding: 0.5rem;

    .pgn__card-image-cap {
      max-width: 99.01px;
    }
  }
  .pgn__card-section {
    padding: 0;
  }
}
