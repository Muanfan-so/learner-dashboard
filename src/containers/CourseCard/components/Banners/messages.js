import { StrictDict } from 'utils';

export const messages = StrictDict({
  auditAccessExpired: {
    id: 'learner-dash.courseCard.banners.auditAccessExpired',
    description: 'Audit access expiration banner message',
    defaultMessage: 'Your audit access to this course has expired.',
  },
  upgradeToAccess: {
    id: 'learner-dash.courseCard.banners.upgradeToAccess',
    description: 'Upgrade prompt for audit-expired learners that can still upgrade',
    defaultMessage: 'Upgrade now to access your course again.',
  },
  findAnotherCourse: {
    id: 'learner-dash.courseCard.banners.findAnotherCourse',
    description: 'Action prompt taking learners to course exploration',
    defaultMessage: 'Find another course',
  },
  upgradeDeadlinePassed: {
    id: 'learner-dash.courseCard.banners.upgradeDeadlinePassed',
    description: 'Audit upgrade deadline passed banner message',
    defaultMessage: 'Your upgrade deadline for this course has passed.  To upgrade, enroll in a session that is farther in the future.',
  },
  exploreCourseDetails: {
    id: 'learner-dash.courseCard.banners.exploreCourseDetails',
    description: 'Action prompt taking learners to course details page',
    defaultMessage: 'Explore course details.',
  },
  certRestricted: {
    id: 'learner-dash.courseCard.banners.certificateRestricted',
    description: 'Restricted certificate warning message',
    defaultMessage: 'Your Certificate of Achievement is being held pending confirmation that the issuance of your Certificate is in compliance with strict U.S. embargoes on Iran, Cuba, Syria, and Sudan.  If you think our system has mistakenly identified you as being connected with one of those countries, please let us know by contacting {supportEmail}.',
  },
  certRefundContactBilling: {
    id: 'learner-dash.courseCard.banners.certificateRefundContactBilling',
    description: 'Message to learners to contact billing for certificate refunds',
    defaultMessage: 'If you would like a refund on your Certificate of Achievement, please contact our billing address {billingEmail}',
  },
  passingGrade: {
    id: 'learner-dash.courseCard.banners.passingGrade',
    description: 'Message to learners with minimum passing grade for the course',
    defaultMessage: 'Grade required to pass the course: {minPassingGrade}',
  },
  notEligibleForCert: {
    id: 'learner-dash.courseCard.banners.notEligibleForCert',
    description: 'Certificate inelligibility message',
    defaultMessage: 'You are not eligible for a certificate.',
  },
  viewGrades: {
    id: 'learner-dash.courseCard.banners.viewGrades',
    description: 'Gradses link text',
    defaultMessage: 'View grades.',
  },
  certReady: {
    id: 'learner-dash.courseCard.banners.certReady',
    description: 'Certificate ready message',
    defaultMessage: 'Congratulations.  Your certificate is ready.',
  },
  viewCertificate: {
    id: 'learner-dash.courseCard.banners.viewCertificate',
    description: 'Certificate link text',
    defaultMessage: 'View Certificate.',
  },
  certMinGrade: {
    id: 'learner-dash.courseCard.banners.certMinGrade',
    description: 'Passing grade requirement message',
    defaultMessage: 'Grade required for a certificate: {minPassingGrade}',
  },
  downloadCertificate: {
    id: 'learner-dash.courseCard.banners.downloadCertificate',
    description: 'Certificate download link text',
    defaultMessage: 'Download Certificate.',
  },
  gradeAndCertReadyAfter: {
    id: 'learner-dash.courseCard.banners.gradseAndCertReadyAfter',
    description: 'Grade and certificate availability date message',
    defaultMessage: 'Your grade and certificate will be ready after {availableDate}.',
  },
  entitlementsUnavailable: {
    id: 'learner-dash.courseCard.banners.entitlementsUnavailable',
    description: 'Entitlements course message when no sessions are available',
    defaultMessage: 'There are no sessions available at the moment.  The course team will create new sessions soon.  If no sessions appear, please contact {emailLink} for information.',
  },
  entitlementsExpiringSoon: {
    id: 'learner-dash.courseCard.banners.entitlementsExpiringSoon',
    description: 'Entitlements course message when the entitlement is expiring soon.',
    defaultMessage: 'You must {selectSessionButton} by {changeDeadline} to access the course.',
  },
  entitlementsExpired: {
    id: 'learner-dash.courseCard.banners.entitlementsExpired',
    description: 'Entitlements course message when the entitlement is expired.',
    defaultMessage: 'You can no longer change sessions.',
  },
  selectSession: {
    id: 'learner-dash.courseCard.banners.selectSession',
    description: 'Entitlements session selection link text',
    defaultMessage: 'select a session',
  },
});

export default messages;
