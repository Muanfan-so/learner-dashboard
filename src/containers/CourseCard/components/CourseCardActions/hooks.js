import { handleEvent } from 'data/services/segment/utils';
import { eventNames } from 'data/services/segment/constants';

export const useCourseCardActionData = () => {
  const trackUpgradeClick = () => {
    handleEvent(eventNames.upgradeCourse, {
      pageName: 'learner_home',
      linkType: 'button',
      linkCategory: 'green_upgrade',
    });
  };

  return {
    trackUpgradeClick,
  };
};

export default useCourseCardActionData;
