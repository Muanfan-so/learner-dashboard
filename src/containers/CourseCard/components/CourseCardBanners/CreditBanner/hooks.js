import { StrictDict } from 'utils';
import { hooks as appHooks } from 'data/redux';

import ApprovedContent from './views/ApprovedContent';
import EligibleContent from './views/EligibleContent';
import MustRequestContent from './views/MustRequestContent';
import PendingContent from './views/PendingContent';
import RejectedContent from './views/RejectedContent';

export const statusComponents = StrictDict({
  pending: PendingContent,
  approved: ApprovedContent,
  rejected: RejectedContent,
});

export const useCreditBannerData = (cardId) => {
  const credit = appHooks.useCardCreditData(cardId);
  const { supportEmail } = appHooks.usePlatformSettingsData();
  if (!credit.isEligible) { return null; }

  const { error, purchased, requestStatus } = credit;
  let ContentComponent = EligibleContent;
  if (purchased) {
    if (requestStatus == null) {
      ContentComponent = MustRequestContent;
    } else if (Object.keys(statusComponents).includes(requestStatus)) {
      ContentComponent = statusComponents[requestStatus];
    }
    // Current behavior is to show Elligible State if unknown request status is returned
  }
  return {
    ContentComponent,
    error,
    supportEmail,
  };
};

export default {
  useCreditBannerData,
};
