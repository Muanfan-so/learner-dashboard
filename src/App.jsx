import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { Helmet } from 'react-helmet';

import { useIntl } from '@edx/frontend-platform/i18n';
import { logError } from '@edx/frontend-platform/logging';
import { initializeHotjar } from '@edx/frontend-enterprise-hotjar';

import { ErrorPage, AppContext } from '@edx/frontend-platform/react';
import Footer from '@edx/frontend-component-footer';
import { Alert } from '@edx/paragon';

import { RequestKeys } from 'data/constants/requests';
import store from 'data/store';
import {
  selectors,
  actions,
  thunkActions,
  hooks as appHooks,
} from 'data/redux';
import LearnerDashboardHeader from 'containers/LearnerDashboardHeader';
import Dashboard from 'containers/Dashboard';
import ZendeskFab from 'components/ZendeskFab';

import track from 'tracking';

import fakeData from 'data/services/lms/fakeData/courses';

import messages from './messages';

import './App.scss';

export const App = () => {
  const dispatch = useDispatch();
  // TODO: made development-only
  const { authenticatedUser } = React.useContext(AppContext);
  const { formatMessage } = useIntl();
  const isFailed = {
    initialize: appHooks.useRequestIsFailed(RequestKeys.initialize),
    refreshList: appHooks.useRequestIsFailed(RequestKeys.refreshList),
  };
  const hasNetworkFailure = isFailed.initialize || isFailed.refreshList;
  const { supportEmail } = appHooks.usePlatformSettingsData();
  React.useEffect(() => {
    if (authenticatedUser?.administrator || process.env.NODE_ENV === 'development') {
      window.loadEmptyData = () => {
        dispatch(thunkActions.app.loadData({ ...fakeData.globalData, courses: [] }));
      };
      window.loadMockData = () => {
        dispatch(thunkActions.app.loadData({
          ...fakeData.globalData,
          courses: [
            ...fakeData.courseRunData,
            ...fakeData.entitlementData,
          ],
        }));
      };
      window.store = store;
      window.selectors = selectors;
      window.actions = actions;
      window.thunkActions = thunkActions;
      window.track = track;
    }
    if (process.env.HOTJAR_APP_ID) {
      try {
        initializeHotjar({
          hotjarId: process.env.HOTJAR_APP_ID,
          hotjarVersion: process.env.HOTJAR_VERSION,
          hotjarDebug: !!process.env.HOTJAR_DEBUG,
        });
      } catch (error) {
        logError(error);
      }
    }
  }, [authenticatedUser, dispatch]);
  return (
    <Router>
      <Helmet>
        <title>{formatMessage(messages.pageTitle)}</title>
      </Helmet>
      <div>
        <LearnerDashboardHeader />
        <main>
          {hasNetworkFailure
            ? (
              <Alert variant="danger">
                <ErrorPage message={formatMessage(messages.errorMessage, { supportEmail })} />
              </Alert>
            ) : (<Dashboard />)}
        </main>
        <Footer logo={process.env.LOGO_POWERED_BY_OPEN_EDX_URL_SVG} />
        <ZendeskFab />
      </div>
    </Router>
  );
};

export default App;
