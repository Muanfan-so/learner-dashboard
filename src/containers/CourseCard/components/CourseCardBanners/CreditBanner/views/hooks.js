import React from 'react';

import { StrictDict } from 'utils';
import { apiHooks } from 'hooks';

import * as module from './hooks';

export const state = StrictDict({
  creditRequestData: (val) => React.useState(val), // eslint-disable-line
});

export const useCreditRequestData = (cardId) => {
  const [requestData, setRequestData] = module.state.creditRequestData(null);
  const createCreditApiRequest = apiHooks.useCreateCreditRequest(cardId);
  const createCreditRequest = (e) => {
    e.preventDefault();
    createCreditApiRequest().then(setRequestData);
  };
  return { requestData, createCreditRequest };
};

export default {
  useCreditRequestData,
};
