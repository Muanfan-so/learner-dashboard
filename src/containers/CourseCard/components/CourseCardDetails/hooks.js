import { useIntl } from '@edx/frontend-platform/i18n';
import { hooks as appHooks } from 'data/redux';

import * as hooks from './hooks';
import messages from './messages';

export const useAccessMessage = ({ cardId }) => {
  const { formatMessage, formatDate } = useIntl();
  const enrollment = appHooks.useCardEnrollmentData(cardId);
  const courseRun = appHooks.useCardCourseRunData(cardId);
  if (!courseRun.isStarted) {
    if (!courseRun.startDate) { return null; }
    const startDate = formatDate(courseRun.startDate);
    return formatMessage(messages.courseStarts, { startDate });
  }
  if (enrollment.isEnrolled) {
    const { isArchived, endDate } = courseRun;
    const {
      accessExpirationDate,
      isAuditAccessExpired,
    } = enrollment;

    if (enrollment.isAudit && accessExpirationDate) {
      return formatMessage(
        isAuditAccessExpired ? messages.accessExpired : messages.accessExpires,
        { accessExpirationDate: formatDate(accessExpirationDate) },
      );
    }
    if (!endDate) { return null; }
    return formatMessage(
      isArchived ? messages.courseEnded : messages.courseEnds,
      { endDate: formatDate(endDate) },
    );
  }
  return null;
};

export const useCardDetailsData = ({ dispatch, cardId }) => {
  const { formatMessage } = useIntl();
  const providerName = appHooks.useCardProviderData(cardId).name;
  const { courseNumber } = appHooks.useCardCourseData(cardId);
  const {
    isEntitlement,
    isFulfilled,
    canChange,
  } = appHooks.useCardEntitlementData(cardId);

  const openSessionModal = appHooks.useUpdateSelectSessionModalCallback(dispatch, cardId);

  return {
    providerName: providerName || formatMessage(messages.unknownProviderName),
    accessMessage: hooks.useAccessMessage({ cardId }),
    isEntitlement,
    isFulfilled,
    canChange,
    openSessionModal,
    courseNumber,
    changeOrLeaveSessionMessage: formatMessage(messages.changeOrLeaveSessionButton),
  };
};

export default useCardDetailsData;
