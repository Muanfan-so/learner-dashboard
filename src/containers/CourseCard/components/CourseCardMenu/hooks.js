import React from 'react';
import { StrictDict } from 'utils';

import track from 'tracking';
import { reduxHooks } from 'hooks';

import * as module from './hooks';

export const state = StrictDict({
  isUnenrollConfirmVisible: (val) => React.useState(val), // eslint-disable-line
  isEmailSettingsVisible: (val) => React.useState(val), // eslint-disable-line
});

export const useUnenrollData = () => {
  const [isVisible, setIsVisible] = module.state.isUnenrollConfirmVisible(false);
  return {
    show: () => setIsVisible(true),
    hide: () => setIsVisible(false),
    isVisible,
  };
};

export const useEmailSettings = () => {
  const [isVisible, setIsVisible] = module.state.isEmailSettingsVisible(false);
  return {
    show: () => setIsVisible(true),
    hide: () => setIsVisible(false),
    isVisible,
  };
};

export const useHandleToggleDropdown = (cardId) => {
  const eventName = track.course.courseOptionsDropdownClicked;
  const trackCourseEvent = reduxHooks.useTrackCourseEvent(eventName, cardId);
  return (isOpen) => {
    if (isOpen) { trackCourseEvent(); }
  };
};

export const useCourseCardMenu = (cardId) => {
  const { courseName } = reduxHooks.useCardCourseData(cardId);
  const { isEnrolled, isEmailEnabled } = reduxHooks.useCardEnrollmentData(cardId);
  const { twitter, facebook } = reduxHooks.useCardSocialSettingsData(cardId);
  const { isMasquerading } = reduxHooks.useMasqueradeData();
  const { isEarned } = reduxHooks.useCardCertificateData(cardId);
  const handleTwitterShare = reduxHooks.useTrackCourseEvent(
    track.socialShare,
    cardId,
    'twitter',
  );
  const handleFacebookShare = reduxHooks.useTrackCourseEvent(
    track.socialShare,
    cardId,
    'facebook',
  );

  const showUnenrollItem = isEnrolled && !isEarned;
  const showDropdown = showUnenrollItem || isEmailEnabled || facebook.isEnabled || twitter.isEnabled;

  return {
    courseName,
    isMasquerading,
    isEmailEnabled,
    showUnenrollItem,
    showDropdown,
    facebook,
    twitter,
    handleTwitterShare,
    handleFacebookShare,
  };
};
