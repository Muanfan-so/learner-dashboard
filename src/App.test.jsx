import React from 'react';
import { shallow } from 'enzyme';
import { Helmet } from 'react-helmet';
import { ErrorPage } from '@edx/frontend-platform/react';

import { BrowserRouter as Router } from 'react-router-dom';

import Footer from '@edx/frontend-component-footer';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Alert } from '@edx/paragon';

import { RequestKeys } from 'data/constants/requests';
import { reduxHooks } from 'hooks';
import Dashboard from 'containers/Dashboard';
import LearnerDashboardHeader from 'containers/LearnerDashboardHeader';
import { ExperimentProvider } from 'ExperimentContext';
import { App } from './App';
import messages from './messages';

jest.mock('@edx/frontend-component-footer', () => 'Footer');

jest.mock('containers/Dashboard', () => 'Dashboard');
jest.mock('containers/LearnerDashboardHeader', () => 'LearnerDashboardHeader');
jest.mock('components/ZendeskFab', () => 'ZendeskFab');
jest.mock('ExperimentContext', () => ({
  ExperimentProvider: 'ExperimentProvider',
}));
jest.mock('./widgets/RecommendationsPaintedDoorBtn/PaintedDoorExperimentContext', () => ({
  PaintedDoorExperimentProvider: 'PaintedDoorExperimentProvider',
}));
jest.mock('data/redux', () => ({
  selectors: 'redux.selectors',
  actions: 'redux.actions',
  thunkActions: 'redux.thunkActions',
}));
jest.mock('hooks', () => ({
  reduxHooks: {
    useRequestIsFailed: jest.fn(),
    usePlatformSettingsData: jest.fn(),
    useLoadData: jest.fn(),
  },
}));
jest.mock('data/store', () => 'data/store');

const loadData = jest.fn();
reduxHooks.useLoadData.mockReturnValue(loadData);

const logo = 'fakeLogo.png';
let el;

const supportEmail = 'test-support-url';
reduxHooks.usePlatformSettingsData.mockReturnValue({ supportEmail });

describe('App router component', () => {
  process.env.LOGO_POWERED_BY_OPEN_EDX_URL_SVG = logo;
  const { formatMessage } = useIntl();
  describe('component', () => {
    const runBasicTests = () => {
      test('snapshot', () => { expect(el).toMatchSnapshot(); });
      it('displays title in helmet component', () => {
        expect(el.find(Helmet).find('title').text()).toEqual(useIntl().formatMessage(messages.pageTitle));
      });
      it('displays learner dashboard header', () => {
        expect(el.find(LearnerDashboardHeader).length).toEqual(1);
      });
      it('wraps the page in a browser router', () => {
        expect(el.find(Router)).toMatchObject(el);
      });
      test('Footer logo drawn from env variable', () => {
        expect(el.find(Footer).props().logo).toEqual(logo);
      });
    };
    describe('no network failure', () => {
      beforeAll(() => {
        reduxHooks.useRequestIsFailed.mockReturnValue(false);
        el = shallow(<App />);
      });
      runBasicTests();
      it('loads dashboard', () => {
        expect(el.find('main')).toMatchObject(shallow(
          <main><ExperimentProvider><Dashboard /></ExperimentProvider></main>,
        ));
      });
    });
    describe('initialize failure', () => {
      beforeAll(() => {
        reduxHooks.useRequestIsFailed.mockImplementation((key) => key === RequestKeys.initialize);
        el = shallow(<App />);
      });
      runBasicTests();
      it('loads error page', () => {
        expect(el.find('main')).toEqual(shallow(
          <main>
            <Alert variant="danger">
              <ErrorPage message={formatMessage(messages.errorMessage, { supportEmail })} />
            </Alert>
          </main>,
        ));
      });
    });
    describe('refresh failure', () => {
      beforeAll(() => {
        reduxHooks.useRequestIsFailed.mockImplementation((key) => key === RequestKeys.refreshList);
        el = shallow(<App />);
      });
      runBasicTests();
      it('loads error page', () => {
        expect(el.find('main')).toEqual(shallow(
          <main>
            <Alert variant="danger">
              <ErrorPage message={formatMessage(messages.errorMessage, { supportEmail })} />
            </Alert>
          </main>,
        ));
      });
    });
  });
});
