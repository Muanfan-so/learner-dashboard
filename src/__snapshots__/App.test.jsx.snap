// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App router component component initialize failure snapshot 1`] = `
<BrowserRouter>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Learner Home
    </title>
  </HelmetWrapper>
  <div>
    <[object Object]>
      <LearnerDashboardHeader />
      <main>
        <Alert
          variant="danger"
        >
          <ErrorPage
            message="If you experience repeated failures, please email support at test-support-url"
          />
        </Alert>
      </main>
    </[object Object]>
    <Footer
      logo="fakeLogo.png"
    />
    <ZendeskFab />
  </div>
</BrowserRouter>
`;

exports[`App router component component no network failure snapshot 1`] = `
<BrowserRouter>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Learner Home
    </title>
  </HelmetWrapper>
  <div>
    <[object Object]>
      <LearnerDashboardHeader />
      <main>
        <ExperimentProvider>
          <Dashboard />
        </ExperimentProvider>
      </main>
    </[object Object]>
    <Footer
      logo="fakeLogo.png"
    />
    <ZendeskFab />
  </div>
</BrowserRouter>
`;

exports[`App router component component refresh failure snapshot 1`] = `
<BrowserRouter>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Learner Home
    </title>
  </HelmetWrapper>
  <div>
    <[object Object]>
      <LearnerDashboardHeader />
      <main>
        <Alert
          variant="danger"
        >
          <ErrorPage
            message="If you experience repeated failures, please email support at test-support-url"
          />
        </Alert>
      </main>
    </[object Object]>
    <Footer
      logo="fakeLogo.png"
    />
    <ZendeskFab />
  </div>
</BrowserRouter>
`;
